js/ui/grid_core/ui.grid_core.editing.js

```diff
@@ -574,13 +575,15 @@ const EditingController = modules.ViewController.inherit((function() {
         getPopupContent: noop,
 
         _needInsertItem: function(change, changeType) {
-            const that = this;
-            const dataSource = that._dataController.dataSource();
-            const scrollingMode = that.option('scrolling.mode');
+            const dataController = this._dataController;
+            const dataSource = dataController.dataSource();
+            const scrollingMode = this.option('scrolling.mode');
             const pageIndex = dataSource.pageIndex();
             const beginPageIndex = dataSource.beginPageIndex ? dataSource.beginPageIndex() : pageIndex;
             const endPageIndex = dataSource.endPageIndex ? dataSource.endPageIndex() : pageIndex;
             const isLastPage = endPageIndex === dataSource.pageCount() - 1;
+            const topItemIndex = dataController.topItemIndex?.();
+            const bottomItemIndex = dataController.bottomItemIndex?.();
```

Comment:
I think it will be better to declare this variables in default block where they are used.